iocage/Jail.py

```diff
@@ -578,11 +591,94 @@ def _stop_failed_jail(
                     logger=self.logger
                 )
         except iocage.errors.IocageException as e:
-            yield jailLaunchEvent.fail(e)
+            yield from jailLaunchEvent.fail_generator(e)
             raise e
 
         yield jailLaunchEvent.end(stdout=stdout)
 
+    def _start_dependant_jails(
+        self,
+        terms: iocage.Filter.Terms,
+        dependant_jails_seen: typing.List['JailGenerator'],
+        event_scope: typing.Optional['iocage.events.Scope']=None
+    ) -> typing.Generator['iocage.events.IocageEvent', None, None]:
+
+        jailDependantsStartEvent = iocage.events.JailDependantsStart(
+            jail=self,
+            scope=event_scope
+        )
+        started_jails: typing.List[JailGenerator] = []
+
+        yield jailDependantsStartEvent.begin()
+        _depends = self.config["depends"]
+        if len(_depends) == 0:
+            yield jailDependantsStartEvent.skip("No dependant jails")
+        else:
+            dependant_jails = sorted(
```

Comment:
This part is in the else clause because the other option, no dependant jails, is much shorter and easier to relate to the condition. Will refactor the code to get rid of the else block.